htdocs/core/modules/expedition/doc/pdf_espadon.modules.php

```diff
@@ -992,9 +992,47 @@ protected function _pagehead(&$pdf, $object, $showaddress, $outputlangs)
 		}
 		//$pdf->Rect($this->marge_gauche, $this->marge_haute, $this->page_largeur-$this->marge_gauche-$this->marge_droite, 30);
 		if (isModEnabled('barcode')) {
-			// TODO Build code bar with function writeBarCode of barcode module for sending ref $object->ref
-			//$pdf->SetXY($this->marge_gauche+3, $this->marge_haute+3);
-			//$pdf->Image($logo,10, 5, 0, 24);
+			require_once DOL_DOCUMENT_ROOT.'/core/modules/barcode/doc/tcpdfbarcode.modules.php';
+			require_once DOL_DOCUMENT_ROOT.'/core/modules/barcode/doc/phpbarcode.modules.php';
+
+			$encoding ='';
+			if (getDolGlobalString('GENBARCODE_BARCODETYPE_THIRDPARTY')) {
+				$sql = 'SELECT rowid, code,coder FROM '.MAIN_DB_PREFIX.'c_barcode_type';
+				$sql .= ' WHERE rowid='.((int) (getDolGlobalString('GENBARCODE_BARCODETYPE_THIRDPARTY')));
+				$sql .= ' AND entity = 1';
+				$resql = $db->query($sql);
+				if ($resql) {
+					$num = $db->num_rows($resql);
+					if ($num > 0) {
+						$obj = $db->fetch_object($resql);
+						$encoding = $obj->code;
+						$classname = "mod".ucfirst($obj->coder);
+					}
+
+					if (class_exists($classname)) {
+						$module = new $classname($db);
+						if ($module->encodingIsSupported($obj->code)) {
+							$result = $module->writeBarCode($object->ref, $encoding);
```

Comment:
To represent url, it is better to use a QR code, you can get it by replacing param $encoding into 'QRCODE'.

Reply:
ok, done